distributed/scheduler.py

```diff
@@ -6064,19 +6064,35 @@ def handle_long_running(
         We stop the task from being stolen in the future, and change task
         duration accounting as if the task has stopped.
         """
+        if worker not in self.workers:
+            logger.debug(
+                "Received long-running signal from unknown worker %s. Ignoring.", worker
+            )
+            return
+
         if key not in self.tasks:
             logger.debug("Skipping long_running since key %s was already released", key)
             return
+
         ts = self.tasks[key]
-        steal = self.extensions.get("stealing")
-        if steal is not None:
-            steal.remove_key_from_stealable(ts)
 
         ws = ts.processing_on
         if ws is None:
             logger.debug("Received long-running signal from duplicate task. Ignoring.")
             return
 
+        if ws.address != worker:
```

Comment:
Ideally there was a more reliable way to verify the requests integrity.

A chain like this

 processing -> long running -> released -> processing (without a long running transition)

that happens on the same worker would still recognize a stale event as valid. However, I doubt this is a relevant scenario in practice.